src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -30,15 +32,20 @@ internal class CompletionHandler : IRequestHandler<LSP.CompletionParams, LSP.Com
         private readonly ImmutableHashSet<char> _csharpTriggerCharacters;
         private readonly ImmutableHashSet<char> _vbTriggerCharacters;
 
+        private readonly CompletionListCache? _completionListCache;
+
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public CompletionHandler(
-            [ImportMany] IEnumerable<Lazy<CompletionProvider, CompletionProviderMetadata>> completionProviders)
+            [ImportMany] IEnumerable<Lazy<CompletionProvider, CompletionProviderMetadata>> completionProviders,
+            CompletionListCache? completionCache = null)
```

Comment:
Ah, the reason I made it optional was because TypeScript uses our completion handler, and I don't think they'll use the CompletionListCache. Though I guess the code could be clearer if the parameter was made non-optional and we just passed in null explicitly from the TypeScript handler shim class?

Reply:
I keep hearing that the typescript shims will be deleted eventually, so yeah, probably makes sense for this code to be nicer, and just pass null from the shim.